tethys_gizmos/static/tethys_gizmos/js/cesium_map_view.js

```diff
@@ -297,39 +299,47 @@ var CESIUM_MAP_VIEW = (function() {
                                             orientation : orientation,
                                             model : model,
                                          });
-        m_viewer.trackedEntity = entity;
+        entity['layer_name'] = m_model_data.layer_name
+        entity['layer_variable'] = m_model_data.layer_variable
+        entity['layer_id'] = m_model_data.layer_id
     }
 
     // Set Cesium primitives
     cesium_load_primitives = function()
     {
         var $map_element = $('#' + m_map_target);
-        var m_primitives_options = [];
-        var raw_primitives_options = $map_element.data('primitives');
-        if(is_empty_or_undefined(raw_primitives_options))
+        m_primitives = $map_element.data('primitives');
+        if(is_empty_or_undefined(m_primitives))
         {
             return;
         }
-        for (let i = 0; i < raw_primitives_options.length; i++) {
-            var method = string_to_function(Object.keys(raw_primitives_options[i])[0]);
-            var primitives = m_viewer.scene.primitives.add(new method(cesium_options(Object.values(raw_primitives_options[i])[0])));
+        for (let m_primitive of m_primitives) {
+            // Cesium Data is stored in options attribute
+            let m_primitive_option = m_primitive.options
+
+            // Other data is stored in data attribute
+            let m_primitive_data = m_primitive.data
```

Comment:
`m_primitive` => `primitive`
`m_primitive_option` => `primitive_option`
`m_primitive_data` => `primitive_data`

Reply:
(don't use `m_` for local variable names.)